l10n_it_edi_extension/models/account_move.py

```diff
@@ -0,0 +1,874 @@
+        ):
+            vals["l10n_edi_it_register_regdate"] = register_regdate
+
+        partner.write(vals)
+        return partner
+
+    def _l10n_it_edi_search_partner(self, company, vat, codice_fiscale, email):
+        partner = super()._l10n_it_edi_search_partner(
+            company, vat, codice_fiscale, email
+        )
+        edi_attachment = self.l10n_it_edi_attachment_id
+        if (
+            edi_attachment
+            and not partner
+            and not self.env.context.get("skip_create_partner")
+        ):
+            try:
+                xml_tree = edi_attachment._decode_edi_l10n_it_edi(
+                    edi_attachment.name, edi_attachment.raw
+                )[0]["xml_tree"]
+            except Exception as e:
+                raise UserError(self.env._("Error parsing XML: %s") % str(e)) from e
+
+            buyer_seller_info = self._l10n_it_buyer_seller_info()
+            partner_info = buyer_seller_info[
+                "seller" if self.is_purchase_document() else "buyer"
+            ]
+
+            partner = self._l10n_it_edi_create_partner(
+                xml_tree, partner_info["section_xpath"], vat, codice_fiscale
+            )
+            if not partner.l10n_edi_it_electronic_invoice_no_contact_update:
+                partner = self._l10n_it_edi_update_partner(
+                    xml_tree, partner_info["section_xpath"], partner
+                )
+
+            if elements_stabile_organizzazione := xml_tree.xpath(
+                partner_info["section_xpath"] + "/StabileOrganizzazione"
+            ):
+                for element_stabile_organizzazione in elements_stabile_organizzazione:
+                    self.write(
+                        {
+                            "l10n_it_edi_stabile_organizzazione_indirizzo": get_text(
+                                element_stabile_organizzazione, ".//Indirizzo"
+                            ),
+                            "l10n_it_edi_stabile_organizzazione_civico": get_date(
+                                element_stabile_organizzazione, ".//NumeroCivico"
+                            ),
+                            "l10n_it_edi_stabile_organizzazione_cap": get_date(
+                                element_stabile_organizzazione, ".//CAP"
+                            ),
+                            "l10n_it_edi_stabile_organizzazione_comune": get_date(
+                                element_stabile_organizzazione, ".//Comune"
+                            ),
+                            "l10n_it_edi_stabile_organizzazione_provincia": get_date(
+                                element_stabile_organizzazione, ".//Provincia"
+                            ),
+                            "l10n_it_edi_stabile_organizzazione_nazione": get_date(
+                                element_stabile_organizzazione, ".//Nazione"
+                            ),
+                        }
+                    )
```

Comment:
C'è solo un nodo `StabileOrganizzazione`, perché quindi un `for` loop? Tra l'altro se ci fossero più nodi i valori dei campi assegnati verrebbero sovrascritti.

Reply:
Aggiunto in https://github.com/Borruso/l10n-italy/pull/33